src/backend/winit.rs

```diff
@@ -178,6 +180,27 @@ where
     ))
 }
 
+/// Handler trait to recieve window-related events to provide a better *nested* experience.
+pub trait WinitEventsHandler {
+    /// The window was resized, can be used to adjust the associated `wayland::output::Output`s mode.
+    fn resized(&mut self, evlh: &mut EventLoopHandle, width: u32, height: u32);
+    /// The window was moved
+    fn moved(&mut self, evlh: &mut EventLoopHandle, x: i32, h: i32);
+    /// The window gained or lost focus
+    fn focus_changed(&mut self, evlh: &mut EventLoopHandle, focused: bool);
+    /// The window needs to be redrawn
+    fn refresh(&mut self, evlh: &mut EventLoopHandle);
+    /// The window's hidpi factor changed
+    fn hidpi_changed(&mut self, evlh: &mut EventLoopHandle, scale: f32);
+}
+
+impl WinitGraphicsBackend {
+    /// Get a reference to the internally used `winit::Window`
+    pub fn window(&self) -> &WinitWindow {
```

Comment:
In favor of full explicitness, I'd rather name this `.winit_window()`.

Reply:
done